plugins/node/opentelemetry-instrumentation-aws-sdk/src/utils.ts

```diff
@@ -72,16 +72,24 @@ export const extractAttributesFromNormalizedRequest = (
   };
 };
 
-export const bindPromise = (
-  target: Promise<any>,
+export const bindPromise = <T = unknown>(
+  target: Promise<T>,
   contextForCallbacks: Context,
   rebindCount = 1
-): Promise<any> => {
+): Promise<T> => {
   const origThen = target.then;
-  target.then = function (onFulfilled, onRejected) {
+  type PromiseThenParameters = Parameters<Promise<T>['then']>;
```

Comment:
😮

Reply:
is it good or bad? :)

Reply:
never had the chance to learn this advanced typescript stuff 😅